ext/com_dotnet/com_olechar.c

```diff
@@ -103,3 +103,23 @@ PHP_COM_DOTNET_API char *php_com_olestring_to_string(OLECHAR *olestring, size_t
 
 	return string;
 }
+
+zend_string *php_com_bstr_to_string(BSTR bstr, int codepage)
+{
+	zend_string *string;
+	uint32_t length = SysStringLen(bstr);
+
+	string = zend_string_alloc(length, 0);
+	length = WideCharToMultiByte(codepage, 0, bstr, length + 1, ZSTR_VAL(string), length + 1, NULL, NULL);
```

Comment:
Won't this truncate the string due to character count / byte count mismatch?

Reply:
Oh, right, thanks!

While writing a test for this, I noticed that `php_com_string_to_olestring()` isn't binary safe, so I fixed that right away.